File: sort_algorithms/quick_sort/javascript/quick_sort.js
const nonSortedArray = [5, 3, 8, 4, 2]
const inputList = [...nonSortedArray]

/**
 * Sorts an array using the quick sort algorithm.
 *
 * @param {number[]} sortedList - The array of numbers to be sorted.
 * @returns {number[]} - The sorted array.
 */

function quick_sort(sortedList) {
    if (sortedList.length <= 1) return sortedList;
    const pivotIdx = Math.floor(sortedList.length / 2)
    const pivot = sortedList[pivotIdx]
    
    const leftPart = [];
    const rightPart = [];

    for (let i = 0; i < sortedList.length; i += 1) {
        if (i === pivotIdx) continue;

        if (sortedList[i] < pivot) {
            leftPart.push(sortedList[i])
        } else {
            rightPart.push(sortedList[i])
        }
    }

  return [...quick_sort(leftPart), pivot, ...quick_sort(rightPart)];
}

const sorted = quick_sort(inputList)

console.log('not sorted array: ', nonSortedArray)
console.log('sorted array: ', sorted)

const outputElement = document.getElementById('output');
outputElement.innerHTML = '';
outputElement.innerHTML = `
  <p>Not sorted array: ${nonSortedArray}</p>
  <p>Sorted array: ${sorted}</p>
  <p>For details, check the console.</p>
`;

File: quick_sort/python/quick_sort.py
def quick_sort(arr):
    """
    Sorts a list of elements using the quick sort algorithm.
    Args:
        arr (List): The list of elements to be sorted.
    Returns:
        List: A new list containing the sorted elements.
    """
    if len(arr) <= 1:
        return arr

    pivot = arr[len(arr) // 2]
    left_part = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right_part = [x for x in arr if x > pivot]
    return quick_sort(left_part) + middle + quick_sort(right_part)

not_sort_array = [5, 3, 8, 4, 2]

print('not_sort_array-->', not_sort_array)
print('sort_array-->', quick_sort(not_sort_array))